python/cli/data_platform_workflows_cli/collect_charm_bases.py

```diff
@@ -21,7 +34,17 @@ def main():
     yaml_data = yaml.safe_load(
         pathlib.Path(args.charm_directory, "charmcraft.yaml").read_text()
     )
-    bases = [index for index, _ in enumerate(yaml_data["bases"])]
+    # GitHub runner for each base
+    runners = []
+    for base in yaml_data["bases"]:
+        # Bases format: https://discourse.charmhub.io/t/charmcraft-bases-provider-support/4713
+        architectures = (base.get("build-on") or base).get("architectures", ["amd64"])
+        assert (
+            len(architectures) == 1
+        ), f"Multiple architectures ({architectures}) in one (charmcraft.yaml) base not supported. Use one base per architecture"
```

Comment:
Does it mean charmcraft.yaml should have:
```
  - name: ubuntu
    channel: "22.04"
    architectures: [amd64]
  - name: ubuntu
    channel: "22.04"
    architectures: [arm64]
```

and not
```
  - name: ubuntu
    channel: "22.04"
    architectures: [amd64, arm64]
```

Reply:
Will we use `architectures: any` for workload-less charms? Or build/release two independent revisions there also?

Reply:
> Will we use `architectures: any` for workload-less charms? Or build/release two independent revisions there also?

If there are binary dependencies (cryptography, psycopg, etc), we will need separate builds even if there is no workload.

Reply:
Sure, the question is about `s3-integrator` or `data-integrator`...
Will we produce more identical revision for different architectures based on each Git commit?

Reply:
It is exactly my question. Yes, we will always forget... newbies/rush/... 
We should create a charmhub improvement request, as today the WEB output us not useful:

![Screenshot from 2024-02-26 14-48-24](https://github.com/canonical/data-platform-workflows/assets/1928266/93026cc9-0b80-47fd-a4bb-d02ebeab583c)

![Screenshot from 2024-02-26 14-47-54](https://github.com/canonical/data-platform-workflows/assets/1928266/6744e623-e202-4e74-a8b5-0c95badff2fc)

Plus this visibility WEB confusion: https://github.com/canonical/charmhub.io/issues/1658


P.S. OK, the rule is: we should always clearly specify independent architectures even for workload-less charms.
Let's try to sell it @delgod on a sync call today. :crossed_fingers: